internal/pkg/gateway/submit_test.go

```diff
@@ -555,67 +555,6 @@ func TestSubmit(t *testing.T) {
 				orderer4Mock.client.(*mocks.ABClient).BroadcastReturns(nil, errors.New("orderer4 fails"))
 			},
 		},
-		{
```

Comment:
did you remove these tests because they fail with your production code changes?

Reply:
yes

Reply:
They're all there. There's enough of them. The only error is when there are no warrants.

Reply:
But, the production code in your change requires a single acknowledgement from an orderer, the previous code required a quorum. Does that not mean you need to change the tests accordingly?

Reply:
quorum will be checked at the BFTDeliverer stage

Reply:
The BFT deliverer has nothing to do with the gateway or the gateway unit tests.

Reply:
The BFT-deliverer relates to the interaction with bft as a second part of the customer. The first part of the client is sending a request to the ordering, in this case it is the gateway.

[edcc12-modsmart.pdf](https://github.com/hyperledger/fabric/files/12489569/edcc12-modsmart.pdf)

```
From Byzantine Consensus to BFT State Machine Replication: A Latency-
Optimal Transformation
.........
B. Client Operation
.........
When a client wants to issue a request to the replicas, it sends a REQUEST message in the format specified
.........
The client waits for at least f + 1 matching replies from different replicas, for the same sequence number, and return the operation result.
```

The client that sends is the gateway or sdk. 
The client that waits the result is BFT-deliverer.

And the client expects computation results, not "accepted for computation" responses. 

Ok. You decided to confuse the client by expecting (n+f+1)/2 such answers. You have the right. But it has nothing to do with bft.

Reply:
In the paper you cite, upon a timeout, a non faulty node forwards the requests **to all replicas**. 

In the SmartBFT library, the request is forwarded **only to the leader**. 

Therefore, it is not sufficient to send the request to f+1, because the non faulty node among them will only send to the leader, and not to everyone.

Reply:
The practice of building factory applications with bft has shown me that sending a transaction to a single node as well will lead to success 99% of the time. Yes, it is better to send to quorum or all (to hit the leader). But even in those rare cases where fewer nodes than the quorum receive the transaction, the transaction is accepted into the block.
This point affects the speed of the application. In my case, the requirement is to complete the transaction as quickly as possible and move on to the next ones. Handling unnecessary fake errors can slow things down.

Well, this is so, from personal cases.